Clinical-Genomics/cg
# Make cg compatible with loqusdb-v2

Use loqusdb-v2 CLI, with structural variant, and profiling feature.

**How to test:**

The tests should be made on hasta

1. install on stage 

    - [x] Install this feature branch on stage
        - $ bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-cg-stage.sh loqusdb-v2
    - [x] Change the loqusdb section in the cg config file to use an empty mongodb database, and add the binary of loqusdb-v2
        - $ bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-servers-stage.sh cg_loqusdb-v2"
    - [x] Make sure loqusdb-v2 is installed in its own environment (S_loqusdb-v2). If not installed do
        - $ bash /home/proj/stage/servers/resources/hasta.scilifelab.se/update-loqusdb-v2-stage.sh
    
2. prepare new loqusdb database
    - [x] Use loqusdb command "$ loqusdb cases" to ensure database is empty
    - [x] use loqusdb command `loqusdb index` to create indices in the new database
    - [x] use loqusdb command `loqusdb profile --variant-file <snv_vcf>` to load SNVs that will be used for profiling

3. prepare cases to upload

    - [x] Find or prepare 5 separate cases in stage and where one case has been sequenced by both WES and WGS (the duplicate case), where all necessary files are available through housekeeper. Cases that can be used:
        - simpleoryx (WES), neededant (WGS) (contain sample duplicates)
        - grownliger
        - amazedraptor
        - vocalape
    - [x] Make sure the cases are unlinked from cg, so that they will be uploaded with "$ cg upload auto". in https://clinical-api-stage.scilifelab.se 
        - in Analysis table, remove 'uploaded at' for each sample
        - For all samples in the cases, remove 'loqusdb_id' in Sample table

4. test new cg feature SVs and sample ID profiling

    - [x] use `$ cg upload auto` to upload the observations
    - [x] check that cases that contain sample duplicates are not loaded by loqusdb
    - [x] assert that all cases that do not contain duplicates are uploaded in loqusdb
    - [x] check so that the '_id' for the cases in loqusdb are the same in status-db

5. Try to upload some cases again

    - [x] for some cases, try to upload the observations again with `cg upload observations --case_id <case_id>`
    - [x] assert that cg exits the expected way 

6. Manually check observations in loqusdb
    - [x] Check the correctness for observation counts. 
        - check for some variants (SNVs and SVs), that the correct observation counts are given in loqusdb. 


**Expected outcome:**

 - [x] All cases are uploaded to loqusdb except the duplicate case (simpleoryx or neededant, depending on what order they are uploaded)

 - [x] The '_id' for the cases in loqusdb should be the same found in status-db for the samples

 - [x] When trying to upload one of the cases again, cg logs that case is already in loqusdb

 - [x] for the chosen variants, check manually that the observation counts are in agreement with what is found in the VCFs

**Risk assessment:**

  - With this code, structural variants from both WGS and WES runs will be uploaded into loqusdb. In a future PR, a filter needs to be made to only upload cases from WGS runs.
  - In production, uploads to loqusdb is made with 'cg upload auto'. If an error would arise from the new loqusdb api, uploads into other databases via 'cg upload auto' might also fail.

**Reviewed:**

 - [x] code approved by @patrikgrenfeldt 
 - [x] test approved by @adrosenbaum 
 - [x] deploy approved by @ingkebil 

Cg now calls a newer version of loqusdb, the cg API itself is backwards compatible. This will motivate a minor version bump.
